src/node/splitid_context.h

```diff
@@ -0,0 +1,1053 @@
+     {Context<
+        ccf::NodeId>::SigningSessionCache::ProtocolState::SUBMITTED_SIGNATURE,
+      "submitted_signature"},
+     {Context<
+        ccf::NodeId>::SigningSessionCache::ProtocolState::COMPARED_SIGNATURE,
+      "compared_signature"}});
+  DECLARE_JSON_TYPE(Context<ccf::NodeId>::SigningSessionCache);
+  DECLARE_JSON_REQUIRED_FIELDS(
+    Context<ccf::NodeId>::SigningSessionCache, protocol_state, r);
+
+  DECLARE_JSON_ENUM(
+    Context<ccf::NodeId>::ResharingSessionCache::ProtocolState,
+    {{Context<
+        ccf::NodeId>::ResharingSessionCache::ProtocolState::SESSION_CREATED,
+      "session_created"},
+     {Context<
+        ccf::NodeId>::ResharingSessionCache::ProtocolState::SUBMITTED_DEAL,
+      "submitted_deal"},
+     {Context<
+        ccf::NodeId>::ResharingSessionCache::ProtocolState::SUBMITTED_RESHARING,
+      "submitted_resharing"},
+     {Context<
+        ccf::NodeId>::ResharingSessionCache::ProtocolState::COMMITMENTS_UPDATED,
+      "commitments_updated"}});
+  DECLARE_JSON_TYPE(Context<ccf::NodeId>::ResharingSessionCache);
+  DECLARE_JSON_REQUIRED_FIELDS(
+    Context<ccf::NodeId>::ResharingSessionCache, protocol_state);
+
+  DECLARE_JSON_ENUM(
+    Context<ccf::NodeId>::SamplingSessionCache::ProtocolState,
+    {
+      {Context<
+         ccf::NodeId>::SamplingSessionCache::ProtocolState::SESSION_CREATED,
+       "session_created"},
+      {Context<
+         ccf::NodeId>::SamplingSessionCache::ProtocolState::SUBMITTED_DEAL,
+       "submitted_deal"},
+      {Context<
+         ccf::NodeId>::SamplingSessionCache::ProtocolState::SUBMITTED_RESHARING,
+       "submitted_resharing"},
+      {Context<
+         ccf::NodeId>::SamplingSessionCache::ProtocolState::SUBMITTED_OPEN_KEY,
+       "submitted_open_key"},
+      {Context<
+         ccf::NodeId>::SamplingSessionCache::ProtocolState::COMPUTED_PUBLIC_KEY,
+       "computed_public_key"},
+    });
+  DECLARE_JSON_TYPE(Context<ccf::NodeId>::SamplingSessionCache);
+  DECLARE_JSON_REQUIRED_FIELDS(
+    Context<ccf::NodeId>::SamplingSessionCache, protocol_state);
+};
+
+namespace ccf
+{
+  class CCFRequestAdapter : public RequestAdapter<ccf::NodeId>
+  {
+  public:
+    CCFRequestAdapter(
+      std::shared_ptr<HTTPNodeClient> node_client, const ccf::NodeId& nid) :
+      RequestAdapter<ccf::NodeId>(),
+      nid(nid),
+      node_client(node_client)
+    {}
+
+    virtual ~CCFRequestAdapter() {}
+
+    virtual bool submit_registration(
+      const std::vector<uint8_t>& public_key) const override
+    {
+      RegisterRPC::In in = {nid, public_key};
+
+      http::Request request(fmt::format(
+        "/{}/{}",
+        ccf::get_actor_prefix(ccf::ActorsType::nodes),
+        "splitid/register"));
+      request.set_header(
+        http::headers::CONTENT_TYPE, http::headervalues::contenttype::JSON);
+
+      auto body = serdes::pack(in, serdes::Pack::Text);
+      request.set_body(&body);
+      return node_client->make_request(request);
```

Comment:
As discussed, this pattern is repeated in other functions in the file, so perhaps refactor?